src/App.tsx

```diff
@@ -1,39 +1,69 @@
+/* eslint-disable no-console */
 import React from 'react';
+// import { render } from 'react-dom';
 import './App.scss';
+// import { event } from 'cypress/types/jquery';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+type State = {
+  clockName: string,
+  hasClock: boolean,
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerID = 0;
+
+  componentDidMount() {
+    window.addEventListener('click', this.clockShow);
+    window.addEventListener('contextmenu', this.clockShow);
+
+    this.timerID = window.setInterval(() => {
+      this.setState(({ clockName: getRandomName() }
+      ));
+    }, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (this.state.hasClock === true
+      && this.state.clockName !== prevState.clockName && prevProps) {
+      console.debug(`Renamed from ${prevState.clockName} to ${this.state.clockName}`);
+    }
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    clearInterval(this.timerID);
+    document.removeEventListener('mousedown', this.clockShow);
```

Comment:
You need to remove the same event listeners you've added. You have 'click' and 'contextmenu', so you need to remove them accordingly.